README.md

```diff
@@ -1,131 +1,2469 @@
+## Overview
+This API client was generated by the [OpenAPI Generator](https://openapi-generator.tech) project.  By using the [OpenAPI-spec](https://www.openapis.org/) from a remote server, you can easily generate an API client.
 
-## Introduction
+- API version: 2.3.2 (2.3)
+- Package version: 3
+- Generator version: 7.8.0
+- Build package: org.openapitools.codegen.languages.GoClientCodegen
 
-`go-nautobot` package provides the Go Bindings to interact with [Nautobot Source of Truth](https://nautobot.readthedocs.io/en/stable/) API. Nautobot provides OpenAPI 3.0 specs and an `api_version` query parameter to specify the `major.minor` version to generate the schema from.
+## Installation
 
-This package is auto-generated from Nautobot, and it comes with its own versioning schema, independent of Nautobot. For more details about versioning, check the [Release Versioning](#release-versioning) section.
+Install the following dependencies:
 
-## Customization
+```sh
+go get github.com/stretchr/testify/assert
+go get golang.org/x/net/context
+```
 
-This package only generates the bindings for the Nautobot Core application, and not for the rich [apps ecosystem](https://docs.nautobot.com/projects/core/en/stable/apps/) around it. It's likely that the bindings in this package are not 100% correspond to your Nautobot environment, as you may have installed some public apps or your own homegrown ones.
+Put the package under your project folder and add the following in import:
 
-Being aware of it, most often than not, you would need to generate your own bindings using the OpenAPI schema provided by your Nautobot deployment (with the installed apps). These are the steps to reproduce it:
+```go
+import nautobot "github.com/nautobot/go-nautobot"
+```
 
-1. Install `oapi-codegen`.
+To use a proxy, set the environment variable `HTTP_PROXY`:
 
-   ```bash
-   $ go install github.com/deepmap/oapi-codegen/cmd/oapi-codegen@latest
-   ```
+```go
+os.Setenv("HTTP_PROXY", "http://proxy_name:proxy_port")
+```
 
-2. Define your `oapi-codegen` config file. You can use the [one in this repository](development/oapi-config.yml).
+## Configuration of Server URL
 
-3. Download the `swagger.yaml` from the API, using the proper minor version:
+Default configuration comes with `Servers` field that contains server objects as defined in the OpenAPI specification.
 
-   ```bash
-   $ wget http://<your_nautobot>/api/swagger.yaml?api_version=<major.minor_version> -O swagger.yaml
-   ```
+### Select Server Configuration
 
-4. Generate the Go bindings
+For using other server than the one defined on index 0 set context value `nautobot.ContextServerIndex` of type `int`.
 
-   ```bash
-   $ oapi-codegen --config oapi-config.yml swagger.yaml
-   ```
+```go
+ctx := context.WithValue(context.Background(), nautobot.ContextServerIndex, 1)
+```
 
-## Release Versioning
```

Comment:
I would like to keep a documentation of the release logic

Reply:
Done